workflows/methylation/methylation-cohort.wdl

```diff
@@ -0,0 +1,361 @@
+
+    if (beta_length > max_length){
+        scatter ( merge_num in range((beta_length / max_length) + 1)){
+            # Get the sublist of beta files
+            scatter ( beta_num in range(max_length)){
+                Int num = (
+                    if merge_num > 0
+                    then beta_num + (merge_num * max_length)
+                    else beta_num
+                )
+                if (num < beta_length){
+                    File bam_list = unfiltered_normalized_beta[num]
+                }
+            }
+        }
+        scatter (iter in range(length(bam_list))){
+            call combine_data as inner_merge { input:
+                unfiltered_normalized_beta = select_all(bam_list[iter]),
+                combined_file_name = iter + ".combined.csv",
+            }
+        }
+
+        call combine_data as final_merge { input:
+            unfiltered_normalized_beta = inner_merge.combined_beta,
+        }
+    }
+
+    if (beta_length < max_length){
+        call combine_data as simple_merge { input:
+            unfiltered_normalized_beta,
+        }
+    }
+
+    call filter_probes { input:
+        beta_values = select_first(
+            [
+                final_merge.combined_beta,
+                simple_merge.combined_beta,
+            ]
+        ),
+    }
+
+    call generate_umap { input:
+        filtered_beta_values = filter_probes.filtered_beta_values,
+    }
+
+    call plot_umap { input:
+        umap = generate_umap.umap,
+    }
+
+    output {
+        File combined_beta = select_first(
+            [
+                final_merge.combined_beta,
+                simple_merge.combined_beta,
+            ]
+        )
+        File filtered_beta = filter_probes.filtered_beta_values
+        File filtered_probes = filter_probes.filtered_probes
+        File umap_embedding = generate_umap.umap
+        File umap_plot = plot_umap.umap_plot
+    }
+}
+
+task combine_data {
+    meta {
+        description: "Combine data from multiple samples"
+        outputs: {
+            combined_beta: "Combined beta values for all samples"
+        }
+    }
+
+    parameter_meta {
+        unfiltered_normalized_beta: "Array of unfiltered normalized beta values for each sample"
+        combined_file_name: "Name of the combined file"
+        addl_memory_gb: "Additional memory to allocate for the task (in GB)"
+    }
+
+    input {
+        Array[File] unfiltered_normalized_beta
+        String combined_file_name = "combined_beta.csv"
+        Int addl_memory_gb = 0
+    }
+
+    Int memory_gb = ceil(size(unfiltered_normalized_beta, "GiB") * 2) + addl_memory_gb
+    Int disk_size_gb = ceil(size(unfiltered_normalized_beta, "GiB") * 2)
+
+    command <<<
+        echo "Combining data"
+        input_files=""
+        for file in ~{sep(" ", unfiltered_normalized_beta)}
+        do
+            #ln -s ~{sep(" ", unfiltered_normalized_beta)} .
+            ln -s $file .
```

Comment:
Why do these need to be localized the CWD? Just looking at the Python, I would think that's unnecessary. Can you add a comment explaining the reason?

Reply:
IMO if you are going to localize the files, the symlinks needs to be removed before execution ends. Otherwise when debugging the WD is going to be full of broken links. I'd rather just skip the localization entirely.